### PR TITLE
use dns resolver to get account key

### PR DESCRIPTION
**Pull request by cctdaniel**

### CODE REVIEW

File: pythclient/utils.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import ast
+import dns.resolver
+from loguru import logger
+from typing import Optional
+
+DEFAULT_VERSION = "v2"
+
+
+def get_key(network: str, type: str, version: str = DEFAULT_VERSION) -> Optional[str]:
```
Comment (dragonfrond) [inline pythclient/utils.py:9]: Please add a comment here for future use:
Retrieving keys via DNS TXT records should not be considered secure and is provided as a convenience only. Accounts should be stored locally and verified before being used for production.

File: pythclient/pythclient.py [hunk 2]
```diff
@@ -13,18 +13,18 @@
 from .solana import SolanaAccount, SolanaClient, SolanaPublicKey, SOLANA_DEVNET_HTTP_ENDPOINT, SOLANA_DEVNET_WS_ENDPOINT, SolanaPublicKeyOrStr
 from .pythaccounts import PythAccount, PythMappingAccount, PythProductAccount, PythPriceAccount
 from . import exceptions, config, ratelimit
+from .utils import get_key
 
-V1_FIRST_MAPPING_ACCOUNT_KEY = "ArppEFcsybCLE8CRtQJLQ9tLv2peGmQoKWFuiUWm4KBP"
-V2_FIRST_MAPPING_ACCOUNT_KEY = "BmA9Z6FjioHJPpjT39QazZyhDRUdZy2ezwx4GiDdE2u2"
-V2_PROGRAM_KEY = "gSbePebfvPy7tRqimPoVecS2UsBvYv46ynrzWocc92s"
+V2_FIRST_MAPPING_ACCOUNT_KEY = get_key("devnet", "mapping")
+V2_PROGRAM_KEY = get_key("devnet", "program")
```
Comment (SEJeff) [inline pythclient/pythclient.py:19]: Do we want to have the side effect like this at import time? Personally, I'd be ok with leaving `first_mapping_account_key` in the `PythClient` class blank and requiring it. Then we can remove these as constants, but make sure to put examples of them elsewhere.

wdyt?

File: pythclient/utils.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        return ""
+    if len(answer) != 1:
+        logger.error("Invalid number of records returned for {}!", url)
+        return ""
+    # Example of the raw_key:
+    #     "program=FsJ3A3u2vn5cTVofAjvy6y5kwABJAqYWpe4975bi2epH"
+    raw_key = ast.literal_eval(list(answer)[0].to_text())
+    # program=FsJ3A3u2vn5cTVofAjvy6y5kwABJAqYWpe4975bi2epH"
+    _, key = raw_key.split("=", 1)
+    return key
```
Comment (SEJeff) [inline pythclient/utils.py:35]: I wrote this code for pyth-observer as a quick and dirty hack and don't consider it very pretty python. Would you mind adding some tests for it? I'd be willing to help you do so if necessary.

**Comment by SEJeff** [comment]
Side note, you'll want to rebase this once #13 is merged.

File: pythclient/utils.py (hunk 3 above)
Comment (cctdaniel) [reply pythclient/utils.py:35]: will write tests for it 🙂

File: tests/test_pyth_client.py [hunk 4]
```diff
@@ -16,14 +16,17 @@
...
+from mock import AsyncMock, Mock
 
 
 # Using constants instead of fixtures because:
 # 1) these values are not expected to be mutated
 # 2) these values are used in get_account_info_resp() and get_program_accounts_resp()
 #    and so if they are passed in as fixtures, the functions will complain for the args
 #    while mocking the respective functions
+V2_FIRST_MAPPING_ACCOUNT_KEY = 'BmA9Z6FjioHJPpjT39QazZyhDRUdZy2ezwx4GiDdE2u2'
+V2_PROGRAM_KEY = 'gSbePebfvPy7tRqimPoVecS2UsBvYv46ynrzWocc92s'
```
Comment (cctdaniel) [inline tests/test_pyth_client.py:28]: declared as constants for the same reason as the below constants -- doesn't seems like there's a need to mock this?

File: tests/test_utils.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+def mock_dns_resolver_resolve(mocker: MockerFixture) -> Mock:
+    mock = Mock()
+    mocker.patch('dns.resolver.resolve', side_effect=mock)
+    return mock
+
+
+def test_utils_get_program_key(mock_dns_resolver_resolve: Mock, get_answer_program: dns.resolver.Answer) -> None:
+    mock_dns_resolver_resolve.return_value = get_answer_program
+    program_key = get_key("devnet", "program")
+    assert program_key == "gSbePebfvPy7tRqimPoVecS2UsBvYv46ynrzWocc92s"
```
Comment (SEJeff) [inline tests/test_utils.py:62]: Data only and very little to no logic. This is a perfect example of a simple and good unit test. Nice 👍

**Review by SEJeff — APPROVED** [review]
Great stuff! As a general rule, fixtures are closer to properties than callables, so you shouldn't prefix the name with `get_`. Better would be something like `get_answer_program()` --> `answer_program()` and `get_answer_mapping()` --> `answer_mapping()`.

**Comment by murmer-blip** [comment]
closes https://github.com/pyth-network/pyth-client-py/issues/2

**Comment by ali-bahjati** [comment]
right, thank you!

**Pull request merged**